src/drt/src/rp/FlexRP_prep.cpp

```diff
@@ -49,6 +49,114 @@ void FlexRP::prep()
     prep_via2viaForbiddenLen(ndr.get());
     prep_viaForbiddenTurnLen(ndr.get());
   }
+  prep_minStepViasCheck();
+}
+
+void FlexRP::prep_minStepViasCheck()
+{
+  auto bottomLayerNum = getDesign()->getTech()->getBottomLayerNum();
+  auto topLayerNum = getDesign()->getTech()->getTopLayerNum();
+  tech_->via2viaMinStepPatches_.resize((topLayerNum - bottomLayerNum) / 2 + 1);
+  for (auto lNum = bottomLayerNum; lNum <= topLayerNum; lNum++) {
+    frLayer* layer = tech_->getLayer(lNum);
+    if (layer->getType() != frLayerTypeEnum::ROUTING) {
+      continue;
+    }
+    if (lNum - 2 < bottomLayerNum || lNum + 2 > topLayerNum)
+      continue;
+    auto minStepCons = layer->getMinStepConstraint();
+    if (!minStepCons)
+      continue;
+    vector<frBox>* patches = &tech_->via2viaMinStepPatches_[lNum / 2 - 1];
+    frViaDef* downVia
+        = getDesign()->getTech()->getLayer(lNum - 1)->getDefaultViaDef();
```

Comment:
This returns null when you have FR_MASTERSLICE / cut / M1 and there is no via associated with the cut layer.  Really you shouldn't even get here but lNum - 2 < bottomLayerNum is true for M1 (a better check is needed).  This leads to the crash in CI when this nullptr is dereferenced.

Reply:
" lNum - 2 < bottomLayerNum is false for M1 (a better check is needed)"
The procedure looks at via junctions, so we need up and down vias. I was assuming there was no via below M1. If my assumption is correct then this check is correct.

I will fix the null pointer.